### PR TITLE
Finalpresentationsetup 

### PR DESCRIPTION
**Pull request by gijsvanderburgh**
First assigment presentation added

### CODE REVIEW

File: src/power_system_simulation/Final_presentation.ipynb [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+   "metadata": {},
+   "source": [
+    "# Final Presentation ACDC\n",
+    "\n",
+    "In this notebook we will walk through examples of this package. The following points are covered: \n",
+    "- Find downstream vertices\n",
+    "- Find alternative edges\n",
+    "- Handle PGM input format\n",
+    "- Raise errors if the data is incorrect. \n",
+    "- Aggreate the power flow results in tables. \n",
```
Comment (MakiVr) [inline src/power_system_simulation/Final_presentation.ipynb:14]: Aggregrate

Comment (MakiVr) [reply src/power_system_simulation/Final_presentation.ipynb:14]: I think

File: src/power_system_simulation/Final_presentation.ipynb [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vertex_ids = [0, 2, 4, 6, 10]  # All unique vertex ids\n",
+    "edge_ids = [1, 3, 5, 7, 8, 9]  # All unique edge ids\n",
+    "edge_vertex_id_pairs = [\n",
```
Comment (MakiVr) [inline src/power_system_simulation/Final_presentation.ipynb:63]: This is the only one where you did not put a comment on what it was. Do you still want to add that?

**Review by BrittElbers — DISMISSED** [review]
The final presentation looks good

File: src/power_system_simulation/Final_presentation.ipynb [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+      ]
+     },
+     "execution_count": 19,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "\"\"\"      \n",
+    "vertex_0 (source) --edge_1(enabled)-- vertex_2 --edge_9(enabled)-- vertex_10\n",
```
Comment (MakiVr) [inline src/power_system_simulation/Final_presentation.ipynb:100]: Looks nice

File: src/power_system_simulation/Final_presentation.ipynb [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+   "metadata": {},
+   "source": [
+    "# Validation\n",
+    "This graph is tested for the following conditions: \n",
+    "1. `vertex_ids` and `edge_ids` should be unique.\n",
+    "    - This function compares the length of all `vertex_ids` to the set of `vertex_ids` and gives an error if they are not the same. It uses the same approach for `edge_ids`.\n",
+    "2. `edge_vertex_id_pairs` should have the same length as `edge_ids`.\n",
+    "    - This function compares the length of the list of `edge_vertex_id_pairs` and `edge_ids`.\n",
+    "3. `edge_vertex_id_pairs` should contain valid `vertex ids`.\n",
+    "    - Using a loop all `edge_vertex_id_pairs` are checked to also be valild `edge_ids`.\n",
```
Comment (MakiVr) [inline src/power_system_simulation/Final_presentation.ipynb:123]: valild=valid

File: src/power_system_simulation/Final_presentation.ipynb [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+    "    source_vertex_id=source_vertex_id,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Example\n",
+    "When for example 2 unique vertex id's are given the following error will be raised: "
```
Comment (MakiVr) [inline src/power_system_simulation/Final_presentation.ipynb:158]: Is there not an error if they are not unique? Or am I thinking of the wrong error?

File: src/power_system_simulation/Final_presentation.ipynb [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+    "- Which (currently disabled) edge can be enabled to ensure that the graph is again fully connected and acyclic?\n",
+    "- Return a list of all alternative edges.\n",
+    "\n",
+    "Our example graph would return the following results:   \n",
+    "        Call find_alternative_edges with disabled_edge_id=1 will return [7]   \n",
+    "        Call find_alternative_edges with disabled_edge_id=3 will return [7, 8]   \n",
+    "        Call find_alternative_edges with disabled_edge_id=5 will return [8]   \n",
+    "        Call find_alternative_edges with disabled_edge_id=9 will return []   \n",
+    "\n",
+    "This function can be used by using the alternative_edges function. "
```
Comment (MakiVr) [inline src/power_system_simulation/Final_presentation.ipynb:246]: This function works by using the alternative_edges function. ?

**Review by MakiVr — CHANGES_REQUESTED** [review]
Some things are just grammar mistakes which I think you should definitely change. Other things I am not sure if I understand correctly what you want to sat and if my alternative is better. So feel free to ignore those.

File: src/power_system_simulation/Final_presentation.ipynb (hunk 1 above)
Comment (gijsvanderburgh) [reply src/power_system_simulation/Final_presentation.ipynb:14]: Fixed

File: src/power_system_simulation/Final_presentation.ipynb (hunk 2 above)
Comment (gijsvanderburgh) [reply src/power_system_simulation/Final_presentation.ipynb:63]: Fixed

File: src/power_system_simulation/Final_presentation.ipynb (hunk 3 above)
Comment (gijsvanderburgh) [reply src/power_system_simulation/Final_presentation.ipynb:100]: Thank you

**Review by MakiVr — APPROVED** [review]
Thank you for the changes :)

**Review by BrittElbers — APPROVED** [review]
Look good!

**Pull request merged**